common/locale_test.go

```diff
@@ -0,0 +1,120 @@
+package common
+
+import (
+	"net/http"
+	"testing"
+
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestExtractLangFromSubDomain(t *testing.T) {
+
+	Convey("Given a request when ExtractLangFromSubDomain is called and subdomain cy. is used", t, func() {
+		req, _ := http.NewRequest("GET", "http://cy.localhost:21800/jobs", nil)
+		lang := "cy"
+		languageToUse := ExtractLangFromSubDomain(req)
+
+		Convey("Then the language 'cy' is returned in a string", func() {
+			So(languageToUse, ShouldEqual, lang)
+		})
+	})
+	Convey("Given a request when ExtractLangFromSubDomain is called and no subdomain is used", t, func() {
+		req, _ := http.NewRequest("GET", "http://localhost:21800/jobs", nil)
+
+		lang := "en"
+		languageToUse := ExtractLangFromSubDomain(req)
+
+		Convey("Then the language 'en' is returned in a string", func() {
+			So(languageToUse, ShouldEqual, lang)
+		})
+	})
+}
+
+func TestExtractLangFromCookie(t *testing.T) {
+
+	Convey("Given a cookie with lang set to 'cy' when ExtractLangFromCookie is called", t, func() {
+		cookie := http.Cookie{Name: "lang", Value: "cy"}
+		lang := "cy"
+		languageToUse := ExtractLangFromCookie(&cookie)
+
+		Convey("Then lang returns a string 'cy'", func() {
+			So(languageToUse, ShouldEqual, lang)
+		})
+	})
+	Convey("Given a cookie with lang set to 'cy' when ExtractLangFromCookie is called", t, func() {
+		cookie := http.Cookie{Name: "lang", Value: "en"}
+		lang := "en"
+		languageToUse := ExtractLangFromCookie(&cookie)
+
+		Convey("Then lang returns a string 'en'", func() {
+			So(languageToUse, ShouldEqual, lang)
+		})
+	})
```

Comment:
Add another test case where the cookie value is something other than `en` or `cy` and verify it uses the `en` default.

Reply:
Good shout